docs/components/concepts/user-task-listeners.md

```diff
@@ -0,0 +1,179 @@
+| :---------- | :------------------------------------------------------------------------------------------------------------ |
+| `eventType` | Specifies the user task event that triggers the listener. Supported values are `assigning` and `completing` . |
+| `type`      | The name of the job type.                                                                                     |
+| `retries`   | The number of retries for the user task listener job.                                                         |
+
+:::note
+If multiple user task listeners of the same `eventType` (such as multiple `assigning` listeners) are defined on the same user task, they are executed sequentially, one after the other, in the order they are defined in the BPMN model.
+:::
+
+## Implement a user task listener
+
+User task listeners are implemented using [job workers](/components/concepts/job-workers.md), similar to execution listeners and service task jobs. The job worker processes the task listener job, applies corrections, and may optionally deny the operation.
+
+See [open a job worker](/apis-tools/java-client-examples/job-worker-open.md) for an example of how to create a job worker and handler that can also process user task listener jobs.
+
+### Accessing user task data
+
+User task-specific data, such as `assignee` and `priority`, is accessible through the job headers of the user task listener job. The following properties can be retrieved using predefined header names from the `Protocol` class:
+
+| Property          | Header name                                       |
+| ----------------- | ------------------------------------------------- |
+| `action`          | `Protocol.USER_TASK_ACTION_HEADER_NAME`           |
+| `assignee`        | `Protocol.USER_TASK_ASSIGNEE_HEADER_NAME`         |
+| `candidateGroups` | `Protocol.USER_TASK_CANDIDATE_GROUPS_HEADER_NAME` |
+| `candidateUsers`  | `Protocol.USER_TASK_CANDIDATE_USERS_HEADER_NAME`  |
+| `dueDate`         | `Protocol.USER_TASK_DUE_DATE_HEADER_NAME`         |
+| `followUpDate`    | `Protocol.USER_TASK_FOLLOW_UP_DATE_HEADER_NAME`   |
+| `formKey`         | `Protocol.USER_TASK_FORM_KEY_HEADER_NAME`         |
+| `userTaskKey`     | `Protocol.USER_TASK_KEY_HEADER_NAME`              |
+| `priority`        | `Protocol.USER_TASK_PRIORITY_HEADER_NAME`         |
+
+Below is an example of accessing the `assignee` value from the headers:
+
+```java
+import io.camunda.zeebe.protocol.Protocol;
+
+final JobHandler userTaskListenerHandler =
+    (jobClient, job) -> {
+        // Access the 'assignee' from the job headers
+        // highlight-start
+        final String assignee = job.getCustomHeaders()
+            .get(Protocol.USER_TASK_ASSIGNEE_HEADER_NAME);
+        // highlight-end
+
+        System.out.println("The assignee for this user task is: " + assignee);
+
+        // remaining job handler logic
+    };
```

Comment:
This example is not accessible for users only depending on `camunda-client-java` (as it doesn't come with `zeebe-protocol` unless you depend on our bom)

```suggestion
final JobHandler userTaskListenerHandler =
    (jobClient, job) -> {
        // Access the 'assignee' from the job headers
        // highlight-start
        final String assignee = job.getCustomHeaders()
            .get("io.camunda.zeebe:assignee");
        // highlight-end

        System.out.println("The assignee for this user task is: " + assignee);

        // remaining job handler logic
    `};`
```